client/src/components/books/books/BooksModal.1.js

```diff
@@ -0,0 +1,349 @@
+      done: false
+    };
+
+    constructor(props) {
+      super(props);
+      this.onClickSubmit = this.onClickSubmit.bind(this);
+      this.onClickOpenBookCover = this.onClickOpenBookCover.bind(this);
+      this.onChangeUploadInput = this.onChangeUploadInput.bind(this);
+      this.onChangeUploadFile = this.onChangeUploadFile.bind(this);
+      this.onClickOk = this.onClickOk.bind(this);
+    }
+
+    onClickSubmit(event) {
+      event.preventDefault();
+      const { coverPhotoPath } = this.state;
+      this.setState({ isLoading: true });
+      if (coverPhotoPath) {
+        cloudinary.config({
+          cloud_name: 'adesege',
+          api_key: '389297596633574',
+          api_secret: 'snmxINvzk2UnTCvcnq_oolOBZ10'
+        });
+
+        cloudinary.uploader.upload(coverPhotoPath, (result) => {
+          console.log(result, '++++++++++++_____________');
+          this.setState({ coverPhotoPath: result.secure_url });
+
+          this.props.saveBooks(this.state)
+            .then(
+              (response) => {
+                const data = response.data;
+
+                this.props.addFlashMessage({
+                  type: 'success',
+                  text: data
+                });
+
+                this.props.addBook(data);
+
+                this.setState({
+                  isLoading: false
+                });
+
+                $('#add-books')
+                  .find('div button.btn[data-dismiss="modal"]')
+                  .trigger('click');
+              },
+              (errors) => {
+                this.setState({
+                  isLoading: false
+                });
+                this.props.addFlashMessage({
+                  type: 'error',
+                  text: errors.response.data
+                });
+              });
+        }, {
+          crop: 'limit',
+          width: 250,
+          height: 435,
+          eager: [
+            { width: 125, height: 218, crop: 'fill', format: 'jpg' }
+          ]
+        });
+      } else {
+        this.props.addFlashMessage({
+          type: 'error',
+          text: 'Upload photo'
+        });
+        this.setState({
+          isLoading: false
+        });
+      }
+    }
+
+    componentDidMount() {
+      const $addBooks = $('#add-books');
+      $addBooks.modal('show');
+
+      $addBooks.on('hide.bs.modal', (e) => {
+        this.context.router.push('/books');
+      });
+    }
+
+    onClickOpenBookCover(event) {
+      event.preventDefault();
+      const $parent = $(event.target).parent().attr('target');
+      const $target = $(`#${$parent}`);
+      $target.click();
+    }
+
+    onChangeUploadInput(event) {
+      const files = $(event.target)[0].files;
+      if (!files.length || !window.FileReader) return; // no file selected, or no FileReader support
+
+      if (/^image/.test(files[0].type)) { // only image file
+        const reader = new FileReader(); // instance of the FileReader
```

Comment:
## Critical
Reading files is most likely going to be a reusable function, so, move it into a separate file and just get back the result from it.